plugin-build/build.gradle.kts

```diff
@@ -0,0 +1,96 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+    kotlin("jvm") version "1.4.32"
+    id("java-gradle-plugin")
+    id("maven-publish")
+    id("org.jlleitschuh.gradle.ktlint") version "10.0.0"
+}
+
+repositories {
+    jcenter()
+    mavenLocal()
+    mavenCentral()
+    google()
+}
+
+dependencies {
+    compileOnly(gradleApi())
+    compileOnly("com.android.tools.build:gradle:4.1.3")
+
+    testImplementation(gradleTestKit())
+    testImplementation("junit:junit:4.13.2")
+    testImplementation("org.jetbrains.kotlin:kotlin-test:1.4.32")
+    testImplementation("com.android.tools.build:gradle:4.1.3")
+}
+
+tasks.withType<KotlinCompile>().configureEach {
+    kotlinOptions {
+        jvmTarget = "1.8"
+        freeCompilerArgs = listOf("-Xopt-in=kotlin.RequiresOptIn")
+    }
+}
+
+gradlePlugin {
+    plugins {
+        register("sentryPlugin") {
+            id = "io.sentry.android.gradle"
+            implementationClass = "io.sentry.android.gradle.SentryPlugin"
+        }
+    }
+}
+
+ktlint {
+    debug.set(false)
+    verbose.set(true)
+    android.set(true)
+    outputToConsole.set(true)
+    ignoreFailures.set(false)
+    enableExperimentalRules.set(true)
+    filter {
+        exclude("**/generated/**")
+        include("**/kotlin/**")
+    }
+}
+
+/* ktlint-disable max-line-length */
+publishing {
+    publications {
+        this.withType(MavenPublication::class.java) {
```

Comment:
do we need:

```
java {
    withJavadocJar()
    withSourcesJar()
}
```

or something like that? otherwise sources and javadocs wont be generated I guess so publishing to MC would fail

Reply:
Added. That's actually not a requirement for Maven Central.
What is instead a requirement is the GPG signing that is not set up here.

Reply:
are you sure? I remember that having javadoc and sources are mandatory files to publish on maven central, they could be empty, but they need to exist.

Reply:
> are you sure?

You were right, they're actually a requirement :) TIL